[BE] Input check for torch.nn.MultiheadAttention

isdanni opened the pull request:
Summary: Check `embed_dim` and `num_heads ` of `torch.nn.MultiheadAttention`.

Test Plan: Please see GitHub Actions.

Differential Revision: D47943134

Fix: #105630

isdanni commented:
@pytorchbot label "topic: not user facing"

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -964,6 +964,8 @@ class MultiheadAttention(Module):
     def __init__(self, embed_dim, num_heads, dropout=0., bias=True, add_bias_kv=False, add_zero_attn=False,
                  kdim=None, vdim=None, batch_first=False, device=None, dtype=None) -> None:
+        if embed_dim <= 0 or num_heads <= 0:
+            raise ValueError("embed_dim and num_heads must be greater than 0")
```
```suggestion
            raise ValueError(f"embed_dim and num_heads must be greater than 0, got embed_dim={embed_dim} and num_heads={num_heads}")
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -964,6 +964,8 @@ class MultiheadAttention(Module):
     def __init__(self, embed_dim, num_heads, dropout=0., bias=True, add_bias_kv=False, add_zero_attn=False,
                  kdim=None, vdim=None, batch_first=False, device=None, dtype=None) -> None:
+        if embed_dim <= 0 or num_heads <= 0:
+            raise ValueError("embed_dim and num_heads must be greater than 0, got embed_dim={embed_dim} and num_heads={num_heads} instead")
```
This should be an f-string
```suggestion
            raise ValueError(f"embed_dim and num_heads must be greater than 0, got embed_dim={embed_dim} and num_heads={num_heads} instead")
```

mikaylagawarecki reviewed (approved):
Thanks!

mikaylagawarecki commented:
@isdanni hey could you fix lint please!

mikaylagawarecki commented:
@pytorchbot merge

isdanni committed:
[BE] Input check for torch.nn.MultiheadAttention (#106363)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/106363

Check `embed_dim` and `num_heads ` of `torch.nn.MultiheadAttention`.

Test Plan: Please see GitHub Actions.

Differential Revision: D47943134

fbshipit-source-id: 46be35e0493972208c64a64e485aaa26a9b1407f

isdanni commented:
@pytorchbot merge
